### ISSUE TITLE
Add support for generating ICU sort keys

### ISSUE THREAD

**Issue by chewbranca** [apache/couchdb#2067]
[NOTE]: # ( ^^ Provide a general summary of the request in the title above. ^^ )

## Summary

ICU allows for the generation of sort keys to facilitate storing keys in a database in ICU sorted order. We currently only expose the collation logic, and not the logic to generate the sort keys. This issue is to add a new function to generate said sort keys.

[NOTE]: # ( Provide a brief overview of what the new feature is all about. )

## Desired Behaviour

[NOTE]: # ( Tell us how the new feature should work. Be specific. )
[TIP]:  # ( Do NOT give us access or passwords to your actual CouchDB! )

Add `couch_util:get_sort_key/1` that takes a binary string and returns a sort key string.

## Possible Solution

[NOTE]: # ( Not required. Suggest how to implement the addition or change. )

## Additional context

This is needed for views in FoundationDB as we cannot call the collate logic to properly order two view rows.

[TIP]:  # ( Why does this feature matter to you? What unique circumstances do you have? )

**Comment by chewbranca** [comment]
Here's an initial version exposing the ICU sort key logic: https://github.com/apache/couchdb/compare/2067-add-get-sort-key

**Comment by chewbranca** [comment]
I iterated on that and ended up with the version in https://github.com/apache/couchdb/commit/015fe58c71555553e913cc40cdb9713aefe5fea3. Unfortunately that encountered a latent bug in the logic around allocating a new buffer when we need to return values larger than 64 characters and it would segfault the VM. Thanks to @davisp for sorting out the fix for that issue in https://github.com/apache/couchdb/commit/33046b6cd38d550aab8780c9356105ee1891f98a. This code is wrapped up and has been brought into the `prototype/fdb-layer` branch in https://github.com/apache/couchdb/commit/f4e75ed32e32f4685a11f993ae416b5b47a719f7. Closing this out.
